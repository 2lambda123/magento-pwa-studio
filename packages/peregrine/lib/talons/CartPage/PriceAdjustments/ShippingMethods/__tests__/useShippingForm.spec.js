import React from 'react';
import { useMutation } from '@apollo/client';

import createTestInstance from '../../../../../util/createTestInstance';
import { useShippingForm } from '../useShippingForm';

jest.mock('@apollo/client', () => ({
    useApolloClient: jest.fn(),
    useMutation: jest.fn().mockReturnValue([
        jest.fn(),
        {
            called: false,
            error: undefined,
            loading: false
        }
    ])
}));

jest.mock('../../../../../context/cart', () => {
    const state = {
        cartId: 'cart123'
    };
    const api = {};
    const useCartContext = jest.fn(() => [state, api]);

    return { useCartContext };
});

const Component = props => {
    const talonProps = useShippingForm(props);
    return <i talonProps={talonProps} />;
};

const props = {
    selectedValues: {},
    setIsCartUpdating: jest.fn(),
    mutations: {},
    queries: {}
};

test('returns correct shape', () => {
    const tree = createTestInstance(<Component {...props} />);
    const { root } = tree;
    const { talonProps } = root.findByType('i').props;

    expect(talonProps).toMatchSnapshot();
});

test('returns errors and loading state from Apollo', () => {
    useMutation.mockReturnValueOnce([
        jest.fn(),
        {
            called: true,
            error: 'Apollo Error',
            loading: true
        }
    ]);

    const tree = createTestInstance(<Component {...props} />);
    const { root } = tree;
    const { talonProps } = root.findByType('i').props;

    expect(talonProps).toMatchSnapshot();
});
