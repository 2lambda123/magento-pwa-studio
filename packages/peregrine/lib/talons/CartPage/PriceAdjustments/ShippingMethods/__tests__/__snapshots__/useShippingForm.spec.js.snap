// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns correct shape 1`] = `
Object {
  "formErrors": Array [
    undefined,
  ],
  "handleOnSubmit": [Function],
  "handleZipChange": [Function],
  "isSetShippingLoading": false,
}
`;

exports[`returns errors and loading state from Apollo 1`] = `
Object {
  "formErrors": Array [
    "Apollo Error",
  ],
  "handleOnSubmit": [Function],
  "handleZipChange": [Function],
  "isSetShippingLoading": true,
}
`;
